{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "074196f8-2b7e-44a9-a6c8-35d7c411065b",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from rdflib import URIRef, Namespace, Literal, Graph\n",
    "import kglab"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "631686ec-ed7a-4763-abd5-24e033630e65",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load datasets\n",
    "genes = pd.read_csv(\"Data/CSV_files/genes_NTDs.csv\",  dtype = {'MONDO_id': str})\n",
    "drugs = pd.read_csv(\"Data/CSV_files/drugs_NTDs.csv\")\n",
    "go_cel = pd.read_csv(\"Data/CSV_files/GO_cellularcomp_NTDs_genes.csv\", dtype = {'go_id': str})\n",
    "go_bp = pd.read_csv(\"Data/CSV_files/GO_biologicalprcoess_NTDs_genes.csv\",  dtype = {'go_id': str})\n",
    "go_mf = pd.read_csv(\"Data/CSV_files/GO_molfunction_NTDs_genes.csv\",  dtype = {'go_id': str})\n",
    "react_path = pd.read_csv(\"Data/CSV_files/Reactomepaths_genes_NTDs.csv\")\n",
    "wiki_path = pd.read_csv(\"Data/CSV_files/Wikipaths_genes_NTDs.csv\")\n",
    "proteins = pd.read_csv(\"Data/CSV_files/NTDs_proteins.csv\")\n",
    "domains = pd.read_csv(\"Data/CSV_files/Domains_NTDs_proteins.csv\")\n",
    "variants = pd.read_csv(\"Data/CSV_files/Variants_NTDs.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "cd0d8d1a-35bf-48dd-97e7-a5c8b3927e0b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Define namespaces and KG\n",
    "namespaces = {\n",
    "    \"rdf\": \"http://www.w3.org/1999/02/22-rdf-syntax-ns#\",\n",
    "    \"rdfs\": \"https://www.w3.org/TR/rdf-schema/#\",\n",
    "    \"xsd\": \"http://www.w3.org/2001/XMLSchema#\",\n",
    "    \"biolink\": \"https://w3id.org/biolink/vocab/\",\n",
    "    \"go\": \"https://bioregistry.io/go:\",\n",
    "    \"mesh\": \"https://bioregistry.io/mesh:\",\n",
    "    \"ncbigene\": \"https://bioregistry.io/ncbigene:\",\n",
    "    \"ensembl\": \"https://bioregistry.io/ensembl:\",\n",
    "    \"hgnc\":\"https://bioregistry.io/hgnc:\",\n",
    "    \"mondo\":\"https://bioregistry.io/mondo:\",\n",
    "    \"medgen\":\"https://bioregistry.io/medgen:\",\n",
    "    \"orphanet\":\"https://bioregistry.io/orphanet:\",\n",
    "    \"drugbank\": \"https://bioregistry.io/drugbank:\",\n",
    "    \"pubchem\":\"https://bioregistry.io/pubchem.compound:\",\n",
    "    \"chembl\":\"https://bioregistry.io/chembl:\",\n",
    "    \"reactome\": \"https://bioregistry.io/reactome:\",\n",
    "    \"wikipath\": \"https://bioregistry.io/wikipathways:\",\n",
    "    \"uniprot\": \"https://bioregistry.io/uniprot:\",\n",
    "    \"refseq\": \"https://bioregistry.io/refseq:\",\n",
    "    \"interpro\": \"https://bioregistry.io/interpro:\",\n",
    "    \"pfam\":\"https://bioregistry.io/pfam:\",\n",
    "    \"pharmgkb_drug\": \"https://bioregistry.io/pharmgkb.drug:\",\n",
    "    \"dbsnp\": \"https://bioregistry.io/dbsnp:\"\n",
    "}\n",
    "\n",
    "kg = kglab.KnowledgeGraph(\n",
    "    name=\"NTDs_kg\",\n",
    "    namespaces=namespaces,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "efda31df-86ab-4e0e-b483-42c82c3bdcd7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add information about genes associated to the three neglected tropical diseases (NTDs)\n",
    "for index, row in genes.iterrows():\n",
    "    # Define the classes\n",
    "    gene_id = row[\"entrezgene_id\"]\n",
    "    disease_ass = row[\"MESH_id\"]\n",
    "    node1 = URIRef(str(kg.get_ns(\"ncbigene\")) + \"{}\".format(gene_id))\n",
    "    node2 = URIRef(str(kg.get_ns(\"mesh\")) + \"{}\".format(disease_ass))\n",
    "        \n",
    "    # Add node types\n",
    "    kg.add(node1, kg.get_ns(\"rdf\").type, kg.get_ns(\"biolink\").Gene)\n",
    "    kg.add(node2, kg.get_ns(\"rdf\").type, kg.get_ns(\"biolink\").Disease)\n",
    "    \n",
    "    # Add predicate between classes\n",
    "    kg.add(node1, kg.get_ns(\"biolink\").gene_associated_with_condition, node2)\n",
    "    \n",
    "    # Add properties for classes\n",
    "    # Labels\n",
    "    gene_symbol = row[\"hgnc_symbol\"]\n",
    "    kg.add(node1, kg.get_ns(\"rdfs\").label, Literal(gene_symbol, datatype=kg.get_ns(\"xsd\").string))\n",
    "    disease_name = row[\"Disease\"]\n",
    "    kg.add(node2, kg.get_ns(\"rdfs\").label, Literal(disease_name, datatype=kg.get_ns(\"xsd\").string))\n",
    "    \n",
    "    # Provenance\n",
    "    kg.add(node1, kg.get_ns(\"biolink\").provided_by, Literal(\"KEGG\", datatype=kg.get_ns(\"xsd\").string))\n",
    "    kg.add(node2, kg.get_ns(\"biolink\").provided_by, Literal(\"KEGG\", datatype=kg.get_ns(\"xsd\").string))\n",
    "    \n",
    "    # Alternate URIs\n",
    "    ensembl_id = row[\"Ensembl_id\"]\n",
    "    node3 = URIRef(str(kg.get_ns(\"ensembl\")) + \"{}\".format(ensembl_id))\n",
    "    kg.add(node1, kg.get_ns(\"biolink\").xref, node3)\n",
    "    hgnc_id = row[\"hgnc_id\"]\n",
    "    node4 = URIRef(str(kg.get_ns(\"hgnc\")) + \"{}\".format(hgnc_id))\n",
    "    kg.add(node1, kg.get_ns(\"biolink\").xref, node4)\n",
    "    mondo_id = row[\"MONDO_id\"]\n",
    "    node5 = URIRef(str(kg.get_ns(\"mondo\")) + \"{}\".format(mondo_id))\n",
    "    kg.add(node2, kg.get_ns(\"biolink\").xref, node5)\n",
    "    medgen_id = row[\"MedGen_id\"]\n",
    "    node6 = URIRef(str(kg.get_ns(\"medgen\")) + \"{}\".format(medgen_id))\n",
    "    kg.add(node2, kg.get_ns(\"biolink\").xref, node6)\n",
    "    orphanet_id = row[\"Orphanet_id\"]\n",
    "    node7 = URIRef(str(kg.get_ns(\"orphanet\")) + \"{}\".format(orphanet_id))\n",
    "    kg.add(node2, kg.get_ns(\"biolink\").xref, node7)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "16d38ba2-2e68-4383-bab8-9bbe5c9a8a26",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add information about proteins encoded by genes \n",
    "for index, row in proteins.iterrows():\n",
    "    # Define the classes\n",
    "    gene_id = row[\"entrezgene_id\"]\n",
    "    uniprot_id = row[\"UniProt_id\"]\n",
    "    node1 = URIRef(str(kg.get_ns(\"ncbigene\")) + \"{}\".format(gene_id))\n",
    "    node2 = URIRef(str(kg.get_ns(\"uniprot\")) + \"{}\".format(uniprot_id))\n",
    "    \n",
    "    # Add node types\n",
    "    kg.add(node2, kg.get_ns(\"rdf\").type, kg.get_ns(\"biolink\").Protein)\n",
    "    \n",
    "    # Add predicate between classes\n",
    "    kg.add(node1, kg.get_ns(\"biolink\").translates_to, node2)\n",
    "    \n",
    "    # Add properties for classes\n",
    "    # Labels\n",
    "    protein_name = row[\"hgnc_symbol\"]\n",
    "    kg.add(node2, kg.get_ns(\"rdfs\").label, Literal(protein_name, datatype=kg.get_ns(\"xsd\").string))\n",
    "    \n",
    "    # Provenance\n",
    "    kg.add(node2, kg.get_ns(\"biolink\").provided_by, Literal(\"Uniprot\", datatype=kg.get_ns(\"xsd\").string))\n",
    "    # Alternate URIs\n",
    "    Ensembl_protein_id = row[\"Ensembl_protein_id\"]\n",
    "    if not pd.isna(Ensembl_protein_id):\n",
    "        node3 = URIRef(str(kg.get_ns(\"ensembl\")) + \"{}\".format(Ensembl_protein_id))\n",
    "        kg.add(node2, kg.get_ns(\"biolink\").xref, node3)\n",
    "    \n",
    "    RefSeq_protein_id = row[\"RefSeq_protein_id\"]\n",
    "    if not pd.isna(RefSeq_protein_id):\n",
    "        node4 = URIRef(str(kg.get_ns(\"refseq\")) + \"{}\".format(RefSeq_protein_id))\n",
    "        kg.add(node2, kg.get_ns(\"biolink\").xref, node4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "6442cd9e-6a22-4819-af66-5eca68267c4f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add information about proteins domains\n",
    "for index, row in domains.iterrows():\n",
    "    # Define the classes\n",
    "    uniprot_id = row[\"UniProt_id\"]\n",
    "    interpro_id = row[\"InterPro_id\"]\n",
    "    node1 = URIRef(str(kg.get_ns(\"uniprot\")) + \"{}\".format(uniprot_id))\n",
    "    node2 = URIRef(str(kg.get_ns(\"interpro\")) + \"{}\".format(interpro_id))\n",
    "    \n",
    "    # Add node types\n",
    "    kg.add(node2, kg.get_ns(\"rdf\").type, kg.get_ns(\"biolink\").ProteinDomain)\n",
    "    \n",
    "    # Add predicate between classes\n",
    "    kg.add(node2, kg.get_ns(\"biolink\").part_of, node1)\n",
    "    \n",
    "    # Add properties for classes\n",
    "    # Labels\n",
    "    domain_name = row[\"interpro_description\"]\n",
    "    kg.add(node2, kg.get_ns(\"rdfs\").label, Literal(domain_name, datatype=kg.get_ns(\"xsd\").string))\n",
    "    \n",
    "    # Provenance\n",
    "    kg.add(node2, kg.get_ns(\"biolink\").provided_by, Literal(\"Interpro\", datatype=kg.get_ns(\"xsd\").string))\n",
    "    \n",
    "    # Alternate URIs\n",
    "    Pfam_id = row[\"Pfam_id\"]\n",
    "    if not pd.isna(Pfam_id):\n",
    "        node3 = URIRef(str(kg.get_ns(\"pfam\")) + \"{}\".format(Pfam_id))\n",
    "        kg.add(node2, kg.get_ns(\"biolink\").xref, node3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "63236171-6778-4d9b-8ad9-17b69d1d4bcd",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add information about drugs to treat the NTDs and their binding target proteins\n",
    "for index, row in drugs.iterrows():\n",
    "    # Define the classes\n",
    "    drug_id = row[\"Drugbank.ID\"]\n",
    "    node = URIRef(str(kg.get_ns(\"drugbank\")) + \"{}\".format(drug_id))\n",
    "    \n",
    "    # Add node types\n",
    "    kg.add(node, kg.get_ns(\"rdf\").type, kg.get_ns(\"biolink\").Drug)\n",
    "    \n",
    "    # Add predicate between classes\n",
    "    target_protein = row[\"UniProt_id\"]\n",
    "    kg.add(node, kg.get_ns(\"biolink\").physically_interacts_with, URIRef(str(kg.get_ns(\"uniprot\")) + \"{}\".format(target_protein)))\n",
    "    \n",
    "    disease_ass = row[\"MESH.ID\"]\n",
    "    kg.add(node, kg.get_ns(\"biolink\").treats, URIRef(str(kg.get_ns(\"mesh\")) + \"{}\".format(disease_ass)))\n",
    "    \n",
    "    # Add properties for classes\n",
    "    # Labels\n",
    "    drug_name = row[\"Drug\"]\n",
    "    kg.add(node, kg.get_ns(\"rdfs\").label, Literal(drug_name, datatype=kg.get_ns(\"xsd\").string))\n",
    "    \n",
    "    # Provenance\n",
    "    kg.add(node, kg.get_ns(\"biolink\").provided_by, Literal(\"Drugbank\", datatype=kg.get_ns(\"xsd\").string))\n",
    "    \n",
    "    # Alternate URIs\n",
    "    PubChem_ID = row[\"PubChem.ID\"]\n",
    "    node3 = URIRef(str(kg.get_ns(\"pubchem\")) + \"{}\".format(PubChem_ID))\n",
    "    kg.add(node, kg.get_ns(\"biolink\").xref, node3)\n",
    "    \n",
    "    ChEMBL_ID = row[\"ChEMBL.ID\"]\n",
    "    if not pd.isna(ChEMBL_ID):\n",
    "        node4 = URIRef(str(kg.get_ns(\"chembl\")) + \"{}\".format(ChEMBL_ID))\n",
    "        kg.add(node, kg.get_ns(\"biolink\").xref, node4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "ff38eb8c-cc85-4cbb-94cb-985007b65274",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add information about cellular component gene ontologies from the genes\n",
    "for index, row in go_cel.iterrows():\n",
    "    # Define the class\n",
    "    gene_id = row[\"entrezgene_id\"]\n",
    "    go_id = row[\"go_id\"]\n",
    "    node1 = URIRef(str(kg.get_ns(\"ncbigene\")) + \"{}\".format(gene_id))\n",
    "    node2 = URIRef(str(kg.get_ns(\"go\")) + \"{}\".format(go_id))\n",
    "    \n",
    "    # Add node type\n",
    "    kg.add(node2, kg.get_ns(\"rdf\").type, kg.get_ns(\"biolink\").CellularComponent)\n",
    "    \n",
    "    # Add predicate between classes\n",
    "    kg.add(node1, kg.get_ns(\"biolink\").active_in, node2)\n",
    "    \n",
    "    # Add properties for classes\n",
    "    # Labels\n",
    "    go_name = row[\"name_1006\"]\n",
    "    kg.add(node2, kg.get_ns(\"rdfs\").label, Literal(go_name, datatype=kg.get_ns(\"xsd\").string))\n",
    "    \n",
    "    # Provenance\n",
    "    kg.add(node2, kg.get_ns(\"biolink\").provided_by, Literal(\"Gene Ontology\", datatype=kg.get_ns(\"xsd\").string))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "854cbbdc-47b7-43a6-aae0-7fc8b7154401",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add information about biological processes gene ontologies from the genes\n",
    "for index, row in go_bp.iterrows():\n",
    "    # Define the class\n",
    "    gene_id = row[\"entrezgene_id\"]\n",
    "    go_id = row[\"go_id\"]\n",
    "    node1 = URIRef(str(kg.get_ns(\"ncbigene\")) + \"{}\".format(gene_id))\n",
    "    node2 = URIRef(str(kg.get_ns(\"go\")) + \"{}\".format(go_id))\n",
    "    \n",
    "    # Add node type\n",
    "    kg.add(node2, kg.get_ns(\"rdf\").type, kg.get_ns(\"biolink\").BiologicalProcess)\n",
    "    \n",
    "    # Add predicate between classes\n",
    "    kg.add(node1, kg.get_ns(\"biolink\").acts_upstream_of , node2)\n",
    "    \n",
    "    # Add properties for classes\n",
    "    # Labels\n",
    "    go_name = row[\"name_1006\"]\n",
    "    kg.add(node2, kg.get_ns(\"rdfs\").label, Literal(go_name, datatype=kg.get_ns(\"xsd\").string))\n",
    "    \n",
    "    # Provenance\n",
    "    kg.add(node2, kg.get_ns(\"biolink\").provided_by, Literal(\"Gene Ontology\", datatype=kg.get_ns(\"xsd\").string))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "05c1e363-502b-4460-ad7e-09e1d3bfb5a5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add information about molecular functions gene ontologies from the genes\n",
    "for index, row in go_mf.iterrows():\n",
    "    # Define the class\n",
    "    gene_id = row[\"entrezgene_id\"]\n",
    "    go_id = row[\"go_id\"]\n",
    "    node1 = URIRef(str(kg.get_ns(\"ncbigene\")) + \"{}\".format(gene_id))\n",
    "    node2 = URIRef(str(kg.get_ns(\"go\")) + \"{}\".format(go_id))\n",
    "    \n",
    "    # Add node type\n",
    "    kg.add(node2, kg.get_ns(\"rdf\").type, kg.get_ns(\"biolink\").MolecularActivity)\n",
    "    \n",
    "    # Add predicate between classes\n",
    "    kg.add(node1, kg.get_ns(\"biolink\").participates_in, node2)\n",
    "    \n",
    "    # Add properties for classes\n",
    "    # Labels\n",
    "    go_name = row[\"name_1006\"]\n",
    "    kg.add(node2, kg.get_ns(\"rdfs\").label, Literal(go_name, datatype=kg.get_ns(\"xsd\").string))\n",
    "    \n",
    "    # Provenance\n",
    "    kg.add(node2, kg.get_ns(\"biolink\").provided_by, Literal(\"Gene Ontology\", datatype=kg.get_ns(\"xsd\").string))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "d71d8c30-c02c-4c17-bf13-35049ecf248b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add information about reactome pathways associated to the genes\n",
    "for index, row in react_path.iterrows():\n",
    "    # Define the class\n",
    "    gene_id = row[\"entrezgene_id\"]\n",
    "    path_id = row[\"reactpathID\"]\n",
    "    node1 = URIRef(str(kg.get_ns(\"ncbigene\")) + \"{}\".format(gene_id))\n",
    "    node2 = URIRef(str(kg.get_ns(\"reactome\")) + \"{}\".format(path_id))\n",
    "    \n",
    "    # Add node type\n",
    "    kg.add(node2, kg.get_ns(\"rdf\").type, kg.get_ns(\"biolink\").Pathway)\n",
    "    \n",
    "    # Add predicate between classes\n",
    "    kg.add(node1, kg.get_ns(\"biolink\").participates_in, node2)\n",
    "    \n",
    "    # Add properties for classes\n",
    "    # Labels\n",
    "    go_name = row[\"description\"]\n",
    "    kg.add(node2, kg.get_ns(\"rdfs\").label, Literal(go_name, datatype=kg.get_ns(\"xsd\").string))\n",
    "    \n",
    "    # Provenance\n",
    "    kg.add(node2, kg.get_ns(\"biolink\").provided_by, Literal(\"Reactome\", datatype=kg.get_ns(\"xsd\").string))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "80933a5c-e159-4c0a-94b2-c3209eaffc4e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add information about pathways from wikipathways associated to the genes\n",
    "for index, row in wiki_path.iterrows():\n",
    "    # Define the class\n",
    "    gene_id = row[\"entrezgene_id\"]\n",
    "    path_id = row[\"wikipathID\"]\n",
    "    node1 = URIRef(str(kg.get_ns(\"ncbigene\")) + \"{}\".format(gene_id))\n",
    "    node2 = URIRef(str(kg.get_ns(\"wikipath\")) + \"{}\".format(path_id))\n",
    "    \n",
    "    # Add node type\n",
    "    kg.add(node2, kg.get_ns(\"rdf\").type, kg.get_ns(\"biolink\").Pathway)\n",
    "    \n",
    "    # Add predicate between classes\n",
    "    kg.add(node1, kg.get_ns(\"biolink\").participates_in, node2)\n",
    "    \n",
    "    # Add properties for classes\n",
    "    # Labels\n",
    "    go_name = row[\"description\"]\n",
    "    kg.add(node2, kg.get_ns(\"rdfs\").label, Literal(go_name, datatype=kg.get_ns(\"xsd\").string))\n",
    "    \n",
    "    # Provenance\n",
    "    kg.add(node2, kg.get_ns(\"biolink\").provided_by, Literal(\"WikiPathways\", datatype=kg.get_ns(\"xsd\").string))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "a85b2ae8-c642-4dda-9c5e-4094c57748ca",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Add information about variants associated to genes\n",
    "for index, row in variants.iterrows():\n",
    "    # Define the classes\n",
    "    gene_id = row[\"entrezgene_id\"]\n",
    "    SNP_id = row[\"variant\"]\n",
    "    drug_id = row[\"PharmGKB_ID\"]\n",
    "    node1 = URIRef(str(kg.get_ns(\"ncbigene\")) + \"{}\".format(gene_id))\n",
    "    node2 = URIRef(str(kg.get_ns(\"dbsnp\")) + \"{}\".format(SNP_id))\n",
    "    node3 = URIRef(str(kg.get_ns(\"pharmgkb_drug\")) + \"{}\".format(drug_id))\n",
    "    \n",
    "    # Add node types\n",
    "    kg.add(node2, kg.get_ns(\"rdf\").type, kg.get_ns(\"biolink\").Snv)\n",
    "    kg.add(node3, kg.get_ns(\"rdf\").type, kg.get_ns(\"biolink\").Drug)\n",
    "    \n",
    "    # Add predicate between classes\n",
    "    kg.add(node2, kg.get_ns(\"biolink\").is_sequence_variant_of, node1)\n",
    "    kg.add(node2, kg.get_ns(\"biolink\").associated_with, node3)\n",
    "    \n",
    "    # Add properties for classes\n",
    "    # Labels\n",
    "    chr_name = row[\"chr_name\"]\n",
    "    kg.add(node2, kg.get_ns(\"biolink\").has_sequence_location, Literal(chr_name, datatype=kg.get_ns(\"xsd\").integer))\n",
    "    \n",
    "    allele = row[\"allele\"]\n",
    "    kg.add(node2, kg.get_ns(\"biolink\").has_attribute, Literal(allele, datatype=kg.get_ns(\"xsd\").string))\n",
    "    \n",
    "    phen = row[\"phenotypes\"]\n",
    "    if not pd.isna(phen):\n",
    "        kg.add(node2, kg.get_ns(\"biolink\").has_phenotype, Literal(phen, datatype=kg.get_ns(\"xsd\").string))\n",
    "        \n",
    "    drug_name = row[\"chemicals\"]\n",
    "    kg.add(node3, kg.get_ns(\"rdfs\").label, Literal(drug_name, datatype=kg.get_ns(\"xsd\").string))\n",
    "    \n",
    "    # Provenance\n",
    "    kg.add(node2, kg.get_ns(\"biolink\").provided_by, Literal(\"PharmGKB\", datatype=kg.get_ns(\"xsd\").string))\n",
    "    kg.add(node3, kg.get_ns(\"biolink\").provided_by, Literal(\"PharmGKB\", datatype=kg.get_ns(\"xsd\").string))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "28749138-5fdc-41a0-82f8-5968ada4c250",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Export kg\n",
    "kg.save_rdf(\"./Data/RDF_graphs/NTDs_kg.ttl\", format=\"ttl\", base=None, encoding=\"utf-8\")\n",
    "kg.save_rdf(\"./Data/RDF_graphs/NTDs_kg.nt\", format=\"nt\", base=None, encoding=\"utf-8\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "bd7adcb5-b7b2-48fe-9e8c-71b6edce8e54",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Graph identifier=N9b348ad66ce04b1292ee45b601aecffc (<class 'rdflib.graph.Graph'>)>"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Load NTDs kg with rdflib\n",
    "g = Graph()\n",
    "g.parse(\"Data/RDF_graphs/NTDs_kg.ttl\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "fe7a5a07-0d56-4c41-91bf-0b9703e901a4",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Graph g has 33892 statements.\n"
     ]
    }
   ],
   "source": [
    "# Print the number of \"triples\" in the Graph\n",
    "print(f\"Graph g has {len(g)} statements.\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
